Data/ByteString/Short/Internal.hs

```diff
@@ -482,6 +487,56 @@ toShortIO (BS fptr len) = do
     touchForeignPtr fptr
     ShortByteString <$> stToIO (unsafeFreezeByteArray mba)
 
+-- | A simple wrapper around 'fromShort' that wraps the strict 'ByteString' as
+-- a one-chunk lazy 'LBS.ByteString'.
+lazyFromShort :: ShortByteString -> LBS.ByteString
+lazyFromShort = LBS.fromStrict . fromShort
+
+-- | /O(n)/. Convert a lazy 'LBS.ByteString' into a 'ShortByteString'.
+--
+-- This makes a copy, so does not retain the input string.  Naturally, best
+-- used only with sufficiently short lazy ByteStrings.
+--
+lazyToShort :: LBS.ByteString -> ShortByteString
+lazyToShort LBS.Empty = empty
+lazyToShort (LBS.Chunk c@(BS _ len) cs) =
+    unsafeDupablePerformIO $ do
+        mba0 <- stToIO (newByteArray len)
+        mba <- lazyToShortIO mba0 0 len c cs
+        ShortByteString <$> stToIO (unsafeFreezeByteArray mba)
+
+-- | Append first and any subsequent chunks of lazy bytestring to the provided
+-- buffer, resizing it if required.  The return value is the final buffer.
+-- Arithmetic is done with care, to avoid undetected integer addition
+-- wrap-around.  While nobody should be copying multi-gigabyte lazy bytestrings
+-- to "short" bytestrings, we should error out if possible, rather than corrupt
+-- memory..
+lazyToShortIO :: MutableByteArray RealWorld      -- ^ result buffer
+              -> Int                             -- ^ space already used
+              -> Int                             -- ^ total space allocated
+              -> ByteString                      -- ^ initial strict chunk
+              -> LBS.ByteString                  -- ^ remaining chunks
+              -> IO (MutableByteArray RealWorld) -- ^ possibly resized result
+lazyToShortIO mba used size c@(BS fptr len) cs
+    -- Safe against overflow since used <= size.
```

Comment:
```suggestion
    -- Safe against overflow since 0 <= used <= size.
```

...or we could just say `let newUsed = checkedAdd "Short.lazyToShort" used len`. (This does have the marginal downside of checking for overflow in the no-new-buffer case.)

Another option is to allow `used+len` to signed-overflow, but check `intToWord newUsed <= intToWord size` since it cannot unsigned-overflow.